### DISCUSSION TITLE
Certificate Request Error - even though DNS in cloudflare is set up correctly

### DISCUSSION THREAD

**Discussion by Gystre** [community/community#44425] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

My domain is [kyleyu.org](https://kyleyu.org) which I registered through namecheap and I am trying to use it as my apex domain for my github pages. My dns is handled by cloudflare and it is correctly setup according to the wiki ([https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site)). The github repository for my website is [https://github.com/Gystre/gystre.github.io](https://github.com/Gystre/gystre.github.io)

# Problem

However for about 2 weeks now, github pages has been showing this error:
![github pages](https://user-images.githubusercontent.com/37854809/212573884-ae17c373-6394-4f9a-883d-dadb8a39df37.png)

When I try to check my website:
![web](https://user-images.githubusercontent.com/37854809/212573941-53a2323b-3186-4e9b-a058-dbdc26fe0d73.png)

Here is my cloudflare dns:
![cloudflare dns](https://user-images.githubusercontent.com/37854809/212573915-900b0e39-ba5c-4a3a-9afe-93439445181a.png)

# What I've tried
* removed and readded the custom domain in github pages
* removed and readded the A records for github pages in the cloudflare dns
* removed and readded the CNAME record for `www` in the cloudflare dns
* force website redeploy by pushing updates to the html files
* cleared my cache
* waited for 2 days

I am very confused why it suddenly broke because it was working fine just last month.

**Comment by yoannchaudet** [comment]
Hello,

If you setup a Proxy in front of GitHub Pages for your domain (i.e. Cloudflare), we will not be able to obtain a TLS certificate on your behalf.

Since Cloudflare is acting as your proxy, you will need to set it up to obtain a TLS certificate for you instead. As soon as DNS propagates, your Pages settings will show an error related to the domain not pointing back at GitHub servers and will just stop trying to request a TLS certificate.

**Reply by Gystre** [reply]
Thank you for your advice. I've managed to figure it out and it was as you said where the ip address that was being propogated wasn't pointing to github servers or something which confused it making it stop generating the TLS certificate.

1. To fix it I set the A record proxy statuses to "DNS Only".
![image](https://user-images.githubusercontent.com/37854809/213059208-7e767541-69aa-4668-a84d-f54fb56a9dc6.png)

2. Then I forced a site rebuild by editing some html and pushed the commit
3. I set the A records back to "Proxied" in the cloudflare dns and after a few seconds the right ip addresses were being propogated
![image](https://user-images.githubusercontent.com/37854809/213060521-c6fd0562-b852-47cc-b0c9-a57d83251bf2.png)
